e2e/lvmtest/lvmtest.go

```diff
@@ -0,0 +1,235 @@
+func deleteVHD(num int, force bool) error {
+
+	for i := 1; i <= num; i++ {
+		prefix := fmt.Sprintf("%s%d/%s", brickPrefix, i, lvmPrefix)
+		vhdPath := fmt.Sprintf("%s_vhd", prefix)
+		devicePath := fmt.Sprintf("%s_loop", prefix)
+		_, err := exec.Command("losetup", "-d", devicePath).Output()
+		if err != nil && !force {
+			return err
+		}
+		if err := os.RemoveAll(vhdPath); err != nil && !force {
+			return err
+		}
+		if err := os.RemoveAll(devicePath); err != nil && !force {
+			return err
+		}
+
+	}
+	return nil
+}
+
+//create given number of virtual hard disk
+func createVHD(num int, size string) error {
+
+	for i := 1; i <= num; i++ {
+		prefix := fmt.Sprintf("%s%d/%s", brickPrefix, i, lvmPrefix)
+		vhdPath := fmt.Sprintf("%s_vhd", prefix)
+		devicePath := fmt.Sprintf("%s_loop", prefix)
+		//TODO replace exec command with syscall.Fallocate
+		_, err := exec.Command(fallocateBin, "-l", size, vhdPath).Output()
+		if err != nil {
+			return err
+		}
+		_, err = exec.Command(mknodBin, "-m", "660", devicePath, "b", "7", strconv.Itoa(i+8)).Output()
+		loosetupCmd := exec.Command("losetup", devicePath, vhdPath)
+		_, err = loosetupCmd.Output()
+		if err != nil {
+			return err
+
+		}
+
+	}
+	return nil
+}
+
+//CreateLvmBricks provides an lvm mount point created using loop back devices
+func CreateLvmBricks(prefix string, num int) ([]string, error) {
+	var brickPath []string
+	brickPrefix = prefix
+	var err error
+	if !verifyLVM() {
+		return brickPath, errors.New("lvm or thinlv is not available on the machine")
+	}
+
+	brickPath, err = createBrickPath(num)
+	if err != nil {
+		return brickPath, err
+	}
+	err = createVHD(num, "300M")
+	if err != nil {
+		return brickPath, err
+	}
+
+	err = createLV(num, "200M", "150M")
+	if err != nil {
+		return brickPath, err
+	}
+	return brickPath, nil
+
+}
+
+//Cleanup will kill all process, and remove mount points
+func Cleanup(baseWorkdir, prefix string, brickCount int) {
+
+	brickPrefix = prefix
+	exec.Command("pkill", "gluster").Output()
+	time.Sleep(3 * time.Second)
+
+	mtabEntries, err := volume.GetMounts()
+	if err != nil {
+		return
+	}
+	for _, m := range mtabEntries {
+		if strings.HasPrefix(m.MntDir, baseWorkdir) {
+
+			//Remove any dangling mount pounts
+			exec.Command("umount", "-f", "-l", m.MntDir).Output()
+		}
+	}
+
+	deleteVHD(brickCount, true)
+	deleteLV(brickCount, true)
+
+	vg := fmt.Sprintf("%s_vg_", lvmPrefix)
+	out, err := exec.Command(lvm.LVSCommand, "--noheadings", "-o", "vg_name").Output()
+	for _, entry := range strings.Split(string(out), "\n") {
+		if strings.HasPrefix(entry, vg) {
+			exec.Command(lvm.RemoveCommand, "-f", entry)
+		}
+	}
+	os.RemoveAll(brickPrefix)
+
+}
+
+//CleanupLvmBricks provides an lvm mount point created using loop back devices
+func CleanupLvmBricks(prefix string, num int) error {
+	var err error
+	brickPrefix = prefix
+	if !verifyLVM() {
+		return errors.New("lvm or thinlv is not available on the machine")
+	}
+	err = deleteVHD(num, false)
```

Comment:
For cases where a function returns error and you're checking only the error, (like here)., following the below pattern will be good.

```go
if err := function(); err != nil {
  // Do something
}
```